delta-lake/src/integration/java/org/apache/iceberg/delta/SnapshotDeltaLakeSparkIntegration.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.delta;
+
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.spark.Spark3Util;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.connector.catalog.CatalogPlugin;
+
+class SnapshotDeltaLakeSparkIntegration {
+
+  private SnapshotDeltaLakeSparkIntegration() {}
+
+  static SnapshotDeltaLakeTable snapshotDeltaLakeTable(
+      SparkSession spark, String newTableIdentifier, String deltaTableLocation) {
+    String ctx = "delta lake snapshot target";
+    CatalogPlugin defaultCatalog = spark.sessionState().catalogManager().currentCatalog();
+    Spark3Util.CatalogAndIdentifier catalogAndIdent =
+        Spark3Util.catalogAndIdentifier(ctx, spark, newTableIdentifier, defaultCatalog);
+    return new BaseSnapshotDeltaLakeTableAction(deltaTableLocation)
```

Comment:
following with the proposal to add a `default()` actions provider, we could then just call that actions provider here so this class can be used as an example for people who wants to invoke the snapshoting logic.